src/tests/tracing/eventpipe/common/EventPipeSessionConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Tracing.Tests.Common
+{
+    internal enum EventPipeSerializationFormat
+    {
+        NetPerf,
+        NetTrace
+    }
+
+    internal class EventPipeSessionConfiguration
+    {
+        public EventPipeSessionConfiguration(int circularBufferSizeMB, EventPipeSerializationFormat format, IEnumerable<EventPipeProvider> providers, bool requestRundown=true)
+        {
+            if (circularBufferSizeMB == 0)
+                throw new ArgumentException($"Buffer size cannot be zero.");
+            if (format != EventPipeSerializationFormat.NetPerf && format != EventPipeSerializationFormat.NetTrace)
+                throw new ArgumentException("Unrecognized format");
+            if (providers == null)
+                throw new ArgumentNullException(nameof(providers));
+
+            CircularBufferSizeInMB = circularBufferSizeMB;
+            Format = format;
+            RequestRundown = requestRundown;
+            _providers = new List<EventPipeProvider>(providers);
+        }
+
+        public bool RequestRundown { get; }
+        public int CircularBufferSizeInMB { get; }
+        public EventPipeSerializationFormat Format { get; }
+
+        public IReadOnlyCollection<EventPipeProvider> Providers => _providers.AsReadOnly();
+
+        private readonly List<EventPipeProvider> _providers;
+
+        public byte[] SerializeV2()
```

Comment:
Not really sure why this is named as such, could use a better name to describe what its purpose is?